### PR TITLE
Feature/report post

### PR DESCRIPTION
**Pull request by abdelrhmanali29**
add report post feature

- [x] update swagger file

- [x] route

- [ ] complete service

### CODE REVIEW

File: m3ntorship-Posts-1.0.0-swagger.json [hunk 1]
```diff
@@ -202,6 +200,50 @@
...
+            }
+          }
+        ],
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "type": "object",
+                "properties": {
+                  "informerId": {
```
Comment (Nilegfx) [inline m3ntorship-Posts-1.0.0-swagger.json]: reporterId

File: m3ntorship-Posts-1.0.0-swagger.json [hunk 2]
```diff
@@ -202,6 +200,50 @@
...
+        },
+        "responses": {
+          "200": {
+            "description": "Reported successfully"
+          },
+          "401": {
+            "$ref": "#/components/responses/UnauthorizedError"
+          },
+          "404": {
+            "description": "Post Not Found"
```
Comment (Nilegfx) [inline m3ntorship-Posts-1.0.0-swagger.json]: not found error

File: m3ntorship-Posts-1.0.0-swagger.json [hunk 3]
```diff
@@ -236,7 +275,7 @@
       }
     },
     "/votes/{imageId}": {
-      "post": {
+      "put": {
```
Comment (Nilegfx) [inline m3ntorship-Posts-1.0.0-swagger.json]: why would you change this!

**Review by Nilegfx — CHANGES_REQUESTED** [review]

File: m3ntorship-Posts-1.0.0-swagger.json (hunk 3 above)
Comment (abdelrhmanali29) [reply m3ntorship-Posts-1.0.0-swagger.json]: based on saif request

Comment (Nilegfx) [reply m3ntorship-Posts-1.0.0-swagger.json]: you both shouldn't do this. a PR has to only contain what it meant to solve and not mixing concerns like this one as it is unclear for a reviewer like me why certain change is made in different areas of the swagger file.

in this case, please separate the swagger file in its own PR for me to review. this PR should be Posts openAPI definitions. 

in the future, I'll close any PR that might be a source of confusion for others or it has too many changes/concerns. do NOT accumulate API specifications as it should be done first thing before jumping into any implementations.

**Pull request closed without merging**